code/game/gamemodes/cult/runes.dm

```diff
@@ -414,238 +309,297 @@ var/list/teleport_other_runes = list()
-		return
-	for(var/mob/living/M in orange(1,src))
-		if(!(M in T.contents) && M.stat == DEAD)
-			potential_sacrifice_mobs.Add(M)
-	if(!potential_sacrifice_mobs.len)
-		user << "<span class='cultitalic'>There are no eligible sacrifices nearby!</span>"
-		log_game("Raise Dead rune failed - no catalyst corpse")
-		return
-	mob_to_sacrifice = input(user, "Choose a corpse to sacrifice.", "Corpse to Sacrifice") as null|anything in potential_sacrifice_mobs
-	if(!Adjacent(user) || !src || qdeleted(src) || user.incapacitated())
-		return
-	for(var/mob/living/M in T.contents)
-		if(M.stat == DEAD)
-			potential_revive_mobs.Add(M)
-	if(rune_in_use)
-		return
-	if(!potential_revive_mobs.len)
-		user << "<span class='cultitalic'>There is no eligible revival target on the rune!</span>"
-		log_game("Raise Dead rune failed - no corpse to revived")
-		return
-	mob_to_revive = input(user, "Choose a corpse to revive.", "Corpse to Revive") as null|anything in potential_revive_mobs
-	if(!Adjacent(user) || !src || qdeleted(src) || user.incapacitated())
-		return
-	revive(mob_to_revive, mob_to_sacrifice, user, T)
+/obj/effect/rune/teleport/invoke(mob/living/user)
+	var/list/potential_runes = list()
+	var/list/teleportnames = list()
+	var/list/duplicaterunecount = list()
+	for(var/R in teleport_runes)
+		var/obj/effect/rune/teleport/T = R
+		var/resultkey = T.listkey
+		if(resultkey in teleportnames)
+			duplicaterunecount[resultkey]++
+			resultkey = "[resultkey] ([duplicaterunecount[resultkey]])"
+		else
+			teleportnames.Add(resultkey)
+			duplicaterunecount[resultkey] = 1
+		if(T != src && (T.z <= ZLEVEL_SPACEMAX))
+			potential_runes[resultkey] = T
 
-	//Begin revival
-/obj/effect/rune/raise_dead/proc/revive(mob/living/mob_to_revive, mob/living/mob_to_sacrifice, mob/living/user, turf/T)
-	if(rune_in_use)
-		return
-	if(!in_range(mob_to_sacrifice,src))
-		user << "<span class='cultitalic'>The sacrificial target has been moved!</span>"
+	if(!potential_runes.len)
+		user << "<span class='warning'>There are no valid runes to teleport to!</span>"
+		log_game("Teleport rune failed - no other teleport runes")
 		fail_invoke()
-		log_game("Raise Dead rune failed - catalyst corpse moved")
 		return
-	if(!(mob_to_revive in T.contents))
-		user << "<span class='cultitalic'>The corpse to revive has been moved!</span>"
+
+	if(user.z > ZLEVEL_SPACEMAX)
+		user << "<span class='cultitalic'>You are not in the right dimension!</span>"
+		log_game("Teleport rune failed - user in away mission")
 		fail_invoke()
-		log_game("Raise Dead rune failed - revival target moved")
 		return
-	if(mob_to_sacrifice.stat != DEAD)
-		user << "<span class='cultitalic'>The sacrificial target must be dead!</span>"
+
+	var/input_rune_key = input(user, "Choose a rune to teleport to.", "Rune to Teleport to") as null|anything in potential_runes //we know what key they picked
+	var/obj/effect/rune/teleport/actual_selected_rune = potential_runes[input_rune_key] //what rune does that key correspond to?
```

Comment:
I'm not sure, but would this runtime if you cancel? The input_rune_key ends up being null and you call "potential_runes[input_rune_key]".